File: src/main/java/ru/ifmo/is/mfl/reports/dto/ReportDto.java
package ru.ifmo.is.mfl.reports.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

import ru.ifmo.is.mfl.common.framework.dto.CrudDto;
import ru.ifmo.is.mfl.comments.dto.CommentDto;
import ru.ifmo.is.mfl.reviews.dto.ReviewDto;
import ru.ifmo.is.mfl.users.dto.UserDto;

import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = true)
public class ReportDto extends CrudDto {
  private int id;
  private UserDto user;
  private ReviewDto review;
  private CommentDto comment;
  private boolean resolved;
  private UserDto resolver;
  private String issue;
  private String text;
  private Instant date;
}

File: client/src/interfaces/report/dto/ReportDto.ts
import { CrudDto } from '@/interfaces/crud/dto/CrudDto';
import { UserDto } from '@/interfaces/user/dto/UserDto';

export interface ReportDto extends CrudDto {
  user: UserDto;
  reviewId?: number;
  commentId?: number;
  resolved: boolean;
  issue: string;
  text: string;
  date: Date;
}
